### ISSUE TITLE
Same slug name on Manufacturers 

### ISSUE THREAD

**Issue by Sivolen** [bb-Ricardo/netbox-sync#79]
I have two device on same name, first Lenova, second LENOVO, and they has same slug name.

2021-07-13 14:20:19,751 - ERROR: NetBox returned: PATCH /api/dcim/manufacturers/109/ Bad Request
2021-07-13 14:20:19,751 - ERROR: NetBox returned body: {'slug': ['manufacturer with this slug already exists.']}
2021-07-13 14:20:19,751 - ERROR: Request Failed for manufacturer. Used data: {'slug': 'lenovo'}
2021-07-13 14:20:19,759 - INFO: Creating new NetBox 'device type' object 'System x3550 M5: -[5463AC1]-' with data: {'tags': [{'name': 'NetBox-synced'}, {'name': 'Source: SOMIAC'}], 'model': 'System x3550 M5: -[5463AC1]-', 'manufacturer': 109, 'slug': 'system-x3550-m5--5463ac1-'}
2021-07-13 14:20:19,956 - INFO: NetBox successfully created device type object 'System x3550 M5: -[5463AC1]-'.
2021-07-13 14:20:19,956 - INFO: Creating new NetBox 'device type' object 'System x3550 M5 -[5463AC1]-' with data: {'tags': [{'name': 'NetBox-synced'}, {'name': 'Source: SOMIAC'}], 'model': 'System x3550 M5 -[5463AC1]-', 'manufacturer': 109, 'slug': 'system-x3550-m5--5463ac1-'}
2021-07-13 14:20:20,001 - ERROR: NetBox returned: POST /api/dcim/device-types/ Bad Request
2021-07-13 14:20:20,001 - ERROR: NetBox returned body: {'non_field_errors': ['The fields manufacturer, slug must make a unique set.']}
2021-07-13 14:20:20,002 - ERROR: Request Failed for device type. Used data: {'tags': [{'name': 'NetBox-synced'}, {'name': 'Source: SOMIAC'}], 'model': 'System x3550 M5 -[5463AC1]-', 'manufacturer': 109, 'slug': 'system-x3550-m5--5463ac1-'}

```

**Comment by bb-Ricardo** [comment]
mhh, interesting. This should not happen 😄 , because all names are meant to be compared lowercase.

**Comment by bb-Ricardo** [comment]
Hey, can you run the script with the option "-l DEBUG2" please. You can send the log to my mail address and I will have look. I can't really figure out what's going here from the few lines above. That would be great.

Thank you

**Comment by Sivolen** [comment]
Hi, file with debug in attach  
[debug.log](https://github.com/bb-Ricardo/netbox-sync/files/7102775/debug.log)

**Comment by bb-Ricardo** [comment]
Hi, I found the issue.

You've got Lenovo twice as manufacturer in your NetBox:
```
2021-09-03 12:26:52,459 - INFO: Manufacturer 'LENOVO' attribute 'slug' changed from 'lenovo1' to 'lenovo'
```

you must have one with slug `lenovo` and another one with `lenovo1` in NetBox.
I would suggest to change the manufacturer for all devices with `lenovo1` to `lenovo` and remove the `lenovo1` manufacturer. Then the sync should be fine.

**Comment by Sivolen** [comment]
but I have another Lenovo server and it has the same slug as LENOVO and I manually created the LENOVO manufacturer with lenovo1 slug

**Comment by Sivolen** [comment]
In Monday I delete manufacturer Lenovo and send log again

**Comment by Sivolen** [comment]
Hi, I renamed the slug of all manufacturers to lenovo1 and lenovo2 and this is what happened 

```
2021-09-06 12:41:19,987 - INFO: Updating NetBox 'manufacturer' object 'LENOVO' with data: {'slug': 'lenovo'}
2021-09-06 12:41:19,988 - DEBUG2: Sending PATCH to 'https://10.0.2.18:443/api/dcim/manufacturers/110/' with data 'b'{"slug": "lenovo"}''.
2021-09-06 12:41:20,109 - DEBUG2: Received HTTP Status 200.
2021-09-06 12:41:20,110 - INFO: Updating NetBox 'manufacturer' object 'Lenovo' with data: {'slug': 'lenovo'}
2021-09-06 12:41:20,110 - DEBUG2: Sending PATCH to 'https://10.0.2.18:443/api/dcim/manufacturers/108/' with data 'b'{"slug": "lenovo"}''.
2021-09-06 12:41:20,198 - DEBUG2: Received HTTP Status 400.
2021-09-06 12:41:20,198 - ERROR: NetBox returned: PATCH /api/dcim/manufacturers/108/ Bad Request
```

All log in attach
[debug.log](https://github.com/bb-Ricardo/netbox-sync/files/7113338/debug.log)

**Comment by bb-Ricardo** [comment]
You need to have **ONE** manufacturer `Lenovo` with the slug `lenovo`.
And devices with the second Lenovo manufacturer entry need to edited to set the manufacturer to  THE ONE `Lenovo` manufacturer.

In the end there must be only **ONE** Lenovo manufacturer. You still seem to have two.

**Comment by Sivolen** [comment]
Yes, I agree, but I have two servers, the first gives the name of the Lenovo manufacturer, the second LENOVO and this information cannot be changed on them. 
I myself have created this manufacturer and servers. 
The script swears only that it cannot create the servers but everything else works.

**Comment by bb-Ricardo** [comment]
Dont't worry about upper and lower case. Just remove one of the two Lenovo manufacturers and then try to sync again.

**Comment by Sivolen** [comment]
Left only Lenovo, errors appear in the logs, but the script works well

**Comment by bb-Ricardo** [comment]
So it now assigns the vendor correctly?

Whot other Errors do you have?

**Comment by Sivolen** [comment]
[log.log](https://github.com/bb-Ricardo/netbox-sync/files/7133497/log.log)
Attempts to create producer LENOVO, log with errors in attachment

**Comment by bb-Ricardo** [comment]
can you send me a screenshot of your "Lenovo" manufacturer entry in NetBox?

**Comment by bb-Ricardo** [comment]
Nevermind. I finally found the bug. Will let you know once I pushed the change.

**Comment by bb-Ricardo** [comment]
can you give the latest "development" branch a try to see if this solves your Problem? Thank you

**Comment by Sivolen** [comment]
Hi! The script works strangely, it marks everything as orphaned and fails.
Log and screenshot in attachment. 
[log.log](https://github.com/bb-Ricardo/netbox-sync/files/7140249/log.log)
![Screenshot from 2021-09-10 10-34-57](https://user-images.githubusercontent.com/7619656/132775897-afea67a4-8532-46b7-b38e-ae0c32d7b454.png)

**Comment by bb-Ricardo** [comment]
a, damn it, this was caused by 98294f7c1bd394598fd96ca6b1d66d59a3e23013

can you try out the `development-fixed` branch. That should work better.

**Comment by Sivolen** [comment]
hi! i tried development-fixed branch. I have an errors but script working better.

[log.log](https://github.com/bb-Ricardo/netbox-sync/files/7150427/log.log)

**Comment by bb-Ricardo** [comment]
thank you for testing. Can you give `fix/manufacturer-slug-issue` a try?

**Comment by Sivolen** [comment]
Hi! i tested this version and i have an error, when create a new devices. Log in attachment
[log.log](https://github.com/bb-Ricardo/netbox-sync/files/7275106/log.log)

**Comment by bb-Ricardo** [comment]
seems to be a similar issue with device types as well, where the "name" and the "slug" field don't match exactly in NetBox.

can you get an api response for the device typ with the slug "system-x3550-m5--5463ac1-"? then maybe I can see why this fails.

But It just needs soma manual fixing in your NetBox and then the problems should be solved.
```
2021-10-04 12:43:00,011 - ERROR: NetBox returned body: {'non_field_errors': ['The fields manufacturer, slug must make a unique set.']}
```
this indicates that the device type already exists but either the slug is different or the manufacturer is a different one. If you fix this once manually then the issues should be gone.

**Comment by Sivolen** [comment]
I changed the devices data manually, becous their name System x3550 M5 -[5463AC1]- and System x3550 M5: -[5463AC1]-. They has same slug name

Manufacturer | Lenovo

**Comment by Sivolen** [comment]
Hi, these devices was created automatically the script. Do you mean transferring all dependencies to one and removing the second?

**Comment by bb-Ricardo** [comment]
I added the ":" to characters which are formatted to "-" in slug name.

Can you give `fix/manufacturer-slug-issue` another try?

**Comment by Sivolen** [comment]
I noticed that the script is trying to create two identical devices. 
```

2021-10-22 10:50:49,893 - INFO: Created new device type object: Lenovo Flex System x240 M5 Compute Node -[9532AC1]
2021-10-22 10:50:54,522 - INFO: Created new device type object: Lenovo Flex System x240 M5 Compute Node -[9532AC1]
2021-10-22 10:50:58,767 - INFO: Created new device type object: Lenovo Flex System x240 M5 Compute Node -[9532B2G]
```
